src/main/osd/osd_elements.c

```diff
@@ -601,6 +606,37 @@ static void osdElementArtificialHorizon(osdElementParms_t *element)
 
     element->drawElement = false;  // element already drawn
 }
+
+static void osdElementUpDownReference(osdElementParms_t *element){
+
+// Up/Down reference feature displays reference points on the OSD at Zenith and Nadir
+
+    if(abs(attitude.values.pitch)>450){ //pitch is greater than +45 or less than -45
+        const float earthUpinBodyFrame[3] = {-rMat[2][0], -rMat[2][1], -rMat[2][2]}; //transforum the up vector to the body frame
+
+        float thetaB; // pitch from body frame to zenith/nadir
+        float psiB; // psi from body frame to zenith/nadir
+        char *symbol[2] = {"U", "D"}; // character buffer
+        int direction;
+
+        if(attitude.values.pitch>0.0){ //nose down
+            thetaB = -earthUpinBodyFrame[2]; // get pitch w/re to nadir (use small angle approx for sine)
+            psiB = -earthUpinBodyFrame[1]; // calculate the yaw w/re to nadir (use small angle approx for sine)
+            direction = DOWN;
+        }
+        else{ // nose up
+            thetaB = earthUpinBodyFrame[2]; // get pitch w/re to zenith (use small angle approx for sine)
+            psiB = earthUpinBodyFrame[1]; // calculate the yaw w/re to zenith (use small angle approx for sine)
+            direction = UP;
+        }
+
+        int posX = element->elemPosX + round(scaleRangef(psiB, -M_PIf / 4, M_PIf / 4, -14, 14));
```

Comment:
This element is spanning almost the entire screen size, making it hard for users to position it, and interfering with most other elements - the effecive area should be limited to make it less obtrusive and easier to handle.

Reply:
Mike, thanks for the feedback.  The symbology is normally out of view until high pitch angles  On this issue, the symbology is only in view when pitch exceeds +/- 45.  I only see it during power loops, snap backs etc.  During normal flight when the pilot would have the time to check other elements the symbology is out of view.  If you think it makes sense, I can bound it from the edges.  Please advise if you think that is needed.

Reply:
Yes, I think it makes sense to add a bounding box to this, just like there is a bounding box for every other element. Things like being able to position this on the screen (required to compensate for camera angle) make it necessary to be able to to position this.